wkcuber/cubing.py

```diff
@@ -156,8 +170,48 @@ def cubing_job(args):
                     ]
 
                 buffer = prepare_slices_for_wkw(slices, num_channels)
+                if downsampling_needed:
+                    logging.info(f"Downsampling buffer of size {buffer.shape}.")
```

Comment:
I'd extract lines 174 to 211 into a separate function (e.g., `downsample_unpadded_data`).